integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodsRestart.java

```diff
@@ -450,12 +566,160 @@ public void testServerPodsRestartByChangingImagePullPolicy() {
+            domainUid, domainNamespace));
+    assertNotNull(domain1, "Got null domain resource");
+    assertNotNull(domain1.getSpec(), domain1 + "/spec is null");
+
+    // get the map with server pods and their original creation timestamps
+    podsWithTimeStamps = getPodsWithTimeStamps();
+
+    String oldVersion = assertDoesNotThrow(()
+        -> getDomainCustomResource(domainUid, domainNamespace).getSpec().getRestartVersion());
+    int newVersion = oldVersion == null ? 1 : Integer.valueOf(oldVersion) + 1;
+
+    logger.info("patch the domain resource with new WebLogic secret, restartVersion and introspectVersion");
+    String patchStr
+        = "["
+        + "{\"op\": \"add\", \"path\": \"/spec/restartVersion\", "
+        + "\"value\": \"" + newVersion + "\"}"
+        + "]";
+    logger.info("Updating domain configuration using patch string: {0}\n", patchStr);
+    V1Patch patch = new V1Patch(patchStr);
+    assertTrue(patchDomainCustomResource(domainUid, domainNamespace, patch, V1Patch.PATCH_FORMAT_JSON_PATCH),
+        "Failed to patch domain");
+
+    OffsetDateTime timestamp = now();
+
     // verify the server pods are rolling restarted and back to ready state
     logger.info("Verifying rolling restart occurred for domain {0} in namespace {1}",
         domainUid, domainNamespace);
     assertTrue(verifyRollingRestartOccurred(podsWithTimeStamps, 1, domainNamespace),
         String.format("Rolling restart failed for domain %s in namespace %s", domainUid, domainNamespace));
 
+    logger.info("verify domain roll starting/pod cycle starting events are logged");
+    checkEvent(opNamespace, domainNamespace, domainUid, DOMAIN_ROLL_STARTING, "Normal", timestamp);
+    checkEvent(opNamespace, domainNamespace, domainUid, POD_CYCLE_STARTING, "Normal", timestamp);
+
+    CoreV1Event event = getEvent(opNamespace, domainNamespace,
+        domainUid, DOMAIN_ROLL_STARTING, "Normal", timestamp);
+    logger.info(Yaml.dump(event));
+    logger.info("verify the event message contains the restartVersion changed message is logged");
+    assertTrue(event.getMessage().contains("restart version"));
+
+    event = getEvent(opNamespace, domainNamespace,
+        domainUid, POD_CYCLE_STARTING, "Normal", timestamp);
+    logger.info(Yaml.dump(event));
+    logger.info("verify the event message contains the restartVersion changed message is logged");
+    assertTrue(event.getMessage().contains("restart version"));
+
+    logger.info("verify domain roll completed event is logged");
+    checkEvent(opNamespace, domainNamespace, domainUid, DOMAIN_ROLL_COMPLETED, "Normal", timestamp);
+
+  }
+
+  /**
+   * Modify the image on the domain resource.
+   * Verify all pods are restarted and back to ready state.
+   * Verifies that the domain roll starting/pod cycle starting events are logged.
+   */
+  @Test
+  @DisplayName("Check restart of pods after image change")
+  public void testRestartWithImageChange() {
```

Comment:
I remember we had a similar test in some other class, which was failing intermittently. You and Maggie worked together to resolve the issue

Reply:
Yes, that won't run if we are running tests in 14.1.*. So I added a similar test here which will run for all wls builds.